src/fcntl.rs

```diff
@@ -800,6 +800,9 @@ pub enum FcntlArg<'a> {
     /// Issue an advisory read async with no copy to user
     #[cfg(apple_targets)]
     F_RDADVISE(libc::radvisory),
+    /// Pre-allocate storage with different policies on fd.
```

Comment:
Do you think we shoud document the reason why a mutable reference is needed here? i.e., there is a OUT field in the associated sturcutre🤔